csrc/device_lower/validation.cpp

```diff
@@ -1137,9 +1137,11 @@ void validateAndConvertIterDomainGrouping(Fusion* fusion) {
       // Remember if a grouped ID is found
       is_grouped = true;
 
-      // Grouping only makes sense for the normal iteration type
+      // Grouping only makes sense for the normal iteration or gather scatter
+      // type
       NVF_CHECK(
-          id->getIterType() == IterType::Iteration,
+          id->getIterType() == IterType::Iteration ||
+              id->getIterType() == IterType::GatherScatter,
```

Comment:
naive question:

Before the code change, we are just going to hit a negative assert here. Does this mean we might as well check that the itertype is compatible when we assign ParallelType::Group to an id?

Reply:
that's the job of `void IterDomain::parallelize(ParallelType t)`. Before this PR, the error msg comes from there.

Reply:
Got you. I didn't realize IterDomain::parallelize is defined inside `csrc/ir/nodes.cpp`. :face_with_head_bandage: